### PR TITLE
[#4206] Don't go to fallback if we have a versioned Python.

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
=====

This tries to fix the constant download of python version, when the remote server does not has the configured version.

Changes
=======

Update paver.sh rules for downloading the fallback version.

When using with normal paver.sh it will never use the fallback version... as by now we should have versioned distribution for all our platforms.

It will only go for fallback when you explicitly use the `./paver.sh get_python` as this is used when creating the distributables for  the serve.

As a drive by, I have updated the python version.

How to try and test the changes
===============================

reviewers: @dumol 

Please let me know if you are happy with this fix.
I am happy to receive any feedback regarding any use-cases which might be harder to follow with this change.

```
diff --git a/paver.conf b/paver.conf
index bedcaa4..55180fb 100644
--- a/paver.conf
+++ b/paver.conf
@@ -1,5 +1,5 @@
 # We are in brink itself, so we don't have to install brink.
 BASE_REQUIREMENTS='paver==1.2.4'
-PYTHON_CONFIGURATION='default@2.7.13.25cf4cf:solaris10@2.7.8.25cf4cf'
+PYTHON_CONFIGURATION='default@2.7.13.25cf4cf-NON-EXITENT:solaris10@2.7.8.25cf4cf'
 BINARY_DIST_URI='http://binary.chevah.com/production'
 PIP_INDEX='http://pypi.chevah.com'
```

for testing I went with the above diff.

If I already have a build folder,  running `paver.sh` should fail as the version from paver.conf does not match the existing one, and the configured doesn't exits.

If I don't have a build folder, if will clean the cache version as is not the required one and will try to download and fail.

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

File: paver.sh [hunk 1]
```diff
@@ -356,10 +379,26 @@ copy_python() {
...
                 # We have a different python installed.
+
+                # Check if we have the to-be-updated version and fail if
+                # it does not exists.
+                set +o errexit
+                test_version_exists "$BINARY_DIST_URI/python"
+                local test_version=$?
+                set -o errexit
+                if [ $test_version -ne 0 ]; then
+                    echo "The build is not at $python_installed_version"
```
Comment (dumol) [inline paver.sh]: Perhaps _The build is not at_ -> _The build is now at_ ?

**Comment by adiroiban** [comment]
> $ ./paver.sh get_python
> ./paver.sh: line 694: $2: unbound variable

I don't know what to say about this. 

The get_python is implemented as a simple function, and it does not has input validation.

get_python should always be called as `./paver.sh get_python OS ARCH`

I don't want to add more bash code to have pretty errors, as it will just make the whole file bigger and harder to read

needs-review

**Comment by dumol** [comment]
OK, my bad, I thought `get_python` should also work as a standalone argument.

Please also check my string suggestion above. Thanks!

**changes-approved**

**Pull request merged**